src/aiida_workgraph/utils/__init__.py

```diff
@@ -580,3 +580,28 @@ def serialize_socket_data(input_socket: Dict[str, Any]) -> None:
         if isinstance(value, TaggedValue):
             value = value.__wrapped__
         input_socket["property"]["value"] = general_serializer(value)
+
+
+def clean_node_links_manager(data: Dict[str, Any]) -> Dict[str, Any]:
+    """Recursively clean a NodeLinksManager to a dictionary representation."""
```

Comment:
```suggestion
    """Resolve `NodeLinksManager` instances into dictionary representation."""
```

This function is not the recursive one.